### ISSUE TITLE
Improve performance in word 2016

### ISSUE THREAD

**Issue by hozosch** [nvaccess/nvda#10452]
### Steps to reproduce:
1. Open a word dokument
2. Navigate through it.

### Actual behavior:
NVDA is rather unresponsive
### Expected behavior:
NVDA should be responding imediately.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2019.2.1
#### Windows version:
10.0.18362 workstation
#### Name and version of other software in use when reproducing the issue:
Microsoft word 2016-365
#### Does the issue still occur after restarting your PC?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, just tried the latest alpha snapshot and I have the same problem.
I know, and am able to confirm, that the responsiveness is better when you enable word UI automation from the advanced settings. However, reporting of page numbers doesn't work then, and formatting information is also not how I was used to in word 2010 and windows 7. All is normal in that regard with UI automation disabled.

**Comment by zstanecic** [comment]
Hi!

1.	This sounds like classic office 2016, not office 2016-365
2.	It is normal that the in-process injection is slower
3.	To see how it slows down, can you please send here the sample document, because this occurs with extremely gigantic documents?

Of course, the braille support slows down office in the process injection more than usual on the alpha snapshots and large documents, thus display model can be unresponsible.

**Comment by hozosch** [comment]
O sorry, it's office 2016-365 in fact. I wasn't using a braille display. It also occurs with most documents I tried it with, and they're mostly not that big.

**Comment by DrSooom** [comment]
@hozosch: Please disable the option "Editor revisions" in the Document formatting Settings and try again.

**Comment by hozosch** [comment]
This didn't change a thing for me. I've read in the changelog of 2019.1 though that the performance was improved and to get best results, switch to draft view. I can confirm that it definitely works when that is enabled. However, the lines are very short. I prefer a layout with long lines, which, currently in word 2016-365, it's web layout. but with UIA disabled, it is very laggy, even with editor revisions disabled.

**Comment by bhavyashah** [comment]
I can echo a strong sense of laggines in my experience of navigating documents in Word 2016 even with NVDA 2021.1. The responsiveness improves considerably with the experimental UIA support enabled, just like @hozosch reports, but that has the substantial cost of incomplete support for formatting and double-reading of lists. Given that the default mode is not UIA and that Word 2016 has been around for some while now, I find it curious that there haven't been more reports of this noticeable latency. I will add that I have had editor revision reporting disabled, and this experience does not rely on the largeness of a document or the sophistication of its formatting.

**Comment by cary-rowen** [comment]
Yes, it’s really lagging, and there is even a worse experience. Sometimes a certain line reported by NVDA is not the actual line, but the line above or below it. This is very distressing. I hope that NVDA can improve the overall performance

**Comment by Adriani90** [comment]
@cary-rowen, @hozosch how is this now with NVDA 2023.1 RC1 or NVDA 2022.4? Please make sure to update MS Office 2016 / 365 to the last build version before reproducing the issue.

**Comment by cary-rowen** [comment]
I rarely have problems with NVDA reporting the current line incorrectly anymore, but the lag is still there.
The delay becomes more obvious as the number of pages in the word document increases.
But the improvement of UIA is obvious.

MS365 2302 Build 16.0.16130.20186)

**Comment by hozosch** [comment]
Yes, with UIA it's much better, which is why I use it primarily now.

**Comment by hozosch** [comment]
Ok, now a more detailed report. Didn't try 2023.1 rc1, but with 2022.4 and microsoft office ltsc standard 2021, the responsiveness with the in process method is also acceptable. This, though, could also be because I'm now running a more powerful system with 16 gb ram and six processor cores, so twelve logical processors. I still prefer UIA in any case, and I'm under the impression that the UIA support got better over time.
